pkg/pillar/cmd/volumemgr/handleverifier.go

```diff
@@ -88,63 +99,120 @@ func MaybeRemoveVerifyImageConfig(ctx *volumemgrContext, objType, imageSha strin
 	}
 	if m.RefCount == 0 {
 		log.Fatalf("MaybeRemoveVerifyImageConfig: Attempting to reduce "+
-			"0 RefCount. Image Details - Name: %s, ImageID: %s, "+
-			"ImageSha256:%s, IsContainer: %t",
-			m.Name, m.ImageID, m.ImageSha256, m.IsContainer)
+			"0 RefCount. Image Details - Name: %s, "+
+			"ImageSha256:%s", m.Name, m.ImageSha256)
 	}
 	m.RefCount -= 1
 	log.Infof("MaybeRemoveVerifyImageConfig: RefCount to %d for %s",
 		m.RefCount, imageSha)
-	if m.RefCount == 0 {
-		unpublishVerifyImageConfig(ctx, objType, m.Key())
-	} else {
-		publishVerifyImageConfig(ctx, objType, m)
-	}
+	publishVerifyImageConfig(ctx, objType, m)
 	log.Infof("MaybeRemoveVerifyImageConfig done for %s", imageSha)
 }
 
+// MaybeDeleteVerifyImageConfig checks the refcount and if it is zero it
+// initiates the delete handshake with the verifier. That handshake occurs
+// after a MaybeRemoveVerifyImageConfig has dropped the refcount to zero
+// thus after:
+// 1. volumemgr publishes the VIC with RefCount=0
+// 2. verifier publishes the VIS with RefCount=0
+// At that point in time MaybeAddVerifyImageConfig can be called to increment
+// the refcount, but if MaybeDeleteVerifyImageConfig is called we proceed with
+// 3. volumemgr publishes the VIC with Expired=true (RefCount=0)
```

Comment:
Just to clarify, instead of waiting for a certain period of time (downloadGCTime) after the refCount handshake (both volumemger and verifier ack that refcount of a verified image is 0) is done, volumemgr
will right away mark VerifierImageCoinfig.Expired = true and eventually delete the image?

Reply:
When volumemgr calls the MaybeDeleteVerifyImageConfig it will immediately have the verifier delete it.
Note that no code currently calls that function; we could decide to call it as soon as an object has been copies into containerd. But there is no timer in verifier with these changes.

Reply:
okay.